### PR TITLE
No need to run CI on source-unrelated files

### PR DESCRIPTION
**Pull request by WhiteBlackGoose**
I intentionally only wrote down just two: those are assumed to change often. We could add all files which don't affect the build to it, but imo it makes maintaining this thing harder.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,8 +4,14 @@ on:
     branches:
       - 'main'
       - 'release/*'
+    paths-ignore:
+      - 'proposals/**'
+      - 'README.md'
```
Comment (Perksey) [inline .github/workflows/build.yml:9]: The README does in fact change the pack output. Probably best to run CI for this.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -4,8 +4,14 @@ on:
     branches:
       - 'main'
       - 'release/*'
+    paths-ignore:
+      - 'proposals/**'
```
Comment (Perksey) [inline .github/workflows/build.yml:8]:
```suggestion
      - 'documentation/**'
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -4,8 +4,14 @@ on:
...
+      - 'README.md'
   pull_request:
-    # none
+    # all branches
+    paths-ignore:
+      - 'proposals/**'
```
Comment (Perksey) [inline .github/workflows/build.yml:13]:
```suggestion
      - 'documentation/**'
```

**Review by Perksey — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (WhiteBlackGoose) [reply .github/workflows/build.yml]: You will, of course, ask, *how* on Earth was I able to make this mistake

File: .github/workflows/build.yml (hunk 1 above)
Comment (WhiteBlackGoose) [reply .github/workflows/build.yml:9]: Oh... um, does it really affect anything?

Comment (Perksey) [reply .github/workflows/build.yml]: Nothing we check for, but the README is bundled in every NuGet package and is fed into MSBuild (see [here](https://github.com/dotnet/Silk.NET/blob/main/build/props/common.props#L46))

**Review by Perksey — APPROVED** [review]

**Pull request merged**